### PR TITLE
Adds `deep` param to Dataframe copy method and restrict it to `False`

### LINKED ISSUE #5950: [FEA] Allow shallow or deep copies of dask dataframes
`Dask.dataframe.copy` is currently strictly shallow, which may cause metadata issues when cuDF/pandas objects are scattered across workers and then downstream operations pass `deep=False` as keyword arguments.

It would be useful to parameterize `copy` to accept the deep keyword, and allow either deep or shallow copies:

https://github.com/dask/dask/blob/33ad832dfc80aee72250fe45d3dfc53ebd987f33/dask/dataframe/core.py#L356-L362

### PR DESCRIPTION
**Pull request by jopasdev**
- [x] Closes #5950
- [x] Tests added / passed
- [x] Passes `black dask` / `flake8 dask` / `isort dask`

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

File: dask/dataframe/tests/test_dataframe.py [hunk 1]
```diff
@@ -3995,10 +4003,6 @@ def test_to_datetime():
         dd.to_datetime(ds.index, infer_datetime_format=True),
         check_divisions=False,
     )
-    assert_eq(
-        pd.to_datetime(s, utc=True),
-        dd.to_datetime(ds, utc=True),
-    )
```
Comment (jsignell) [inline dask/dataframe/tests/test_dataframe.py:4001]: This change seems unrelated.

File: dask/dataframe/core.py [hunk 2]
```diff
@@ -404,12 +404,23 @@ def __getstate__(self):
...
         """Make a copy of the dataframe
 
         This is strictly a shallow copy of the underlying computational graph.
         It does not affect the underlying data
+
+        Parameters
+        ----------
+        deep: boolean, default False
+            The deep value must be `False` and it is declared as a parameter just for
+            compatibility with third-party libraries like cuDF operation
```
Comment (jsignell) [inline dask/dataframe/core.py:417]:
```suggestion
        deep : boolean, default False
            The deep value must be `False` and it is declared as a parameter just for
            compatibility with third-party libraries like cuDF
```

File: dask/dataframe/tests/test_dataframe.py (hunk 1 above)
Comment (jopasdev) [reply dask/dataframe/tests/test_dataframe.py:4001]: Yeah, I didn't change it. maybe a outdated fork

**Comment by jsignell** [comment]
ok to test

**Comment by jsignell** [comment]
@jopasdev looks like you are just missing some linting

**Comment by jsignell** [comment]
Looks good to me!
